db/migrate/20200611102258_add_unique_reference_to_user.rb

```diff
@@ -0,0 +1,6 @@
+class AddUniqueReferenceToUser < ActiveRecord::Migration[6.0]
+  def change
+    add_column :users, :unique_user_reference, :string, null: false
+    add_index :users, :unique_user_reference, unique: true
```

Comment:
```suggestion
   add_column :users, :unique_user_reference, :uuid, default: 'uuid_generate_v4()', index: true
```

i'd be tempted to just call the column/field `uuid` as well
```suggestion
   add_column :users, :uuid, :uuid, default: 'uuid_generate_v4()', index: true
```

Reply:
will need to enable a postgres extension if this is not already enabled

https://www.postgresql.org/docs/11/uuid-ossp.html